frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,368 @@
+/* eslint-disable no-undef */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+
+import { connectToPlural } from '../../kinds';
+import { k8sCreate, K8sKind, referenceFor } from '../../module/k8s';
+import { ButtonBar, history, LoadingBox, NameValueEditor, resourceObjPath, RequestSizeInput } from '../utils';
+import { ListDropdown } from '../RBAC/bindings';
+import { RadioInput } from '../radio';
+import { Checkbox } from '../checkbox';
+
+const StorageClassDropdown: React.SFC<StorageClassDropdownProps> = props => {
+  const kind = 'StorageClass';
+  const resources = [{ kind }];
+  const { selectedKey, required, namespace } = props;
+  return (
+    <ListDropdown
+      {...props}
+      desc="Storage Classes"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select storage class"
+      selectedKey={selectedKey}
+      required={required}
+      namespace={namespace}
+      name={name}
+    />
+  );
+};
+// This form is done a little odd since it is used in both its own page and as
+// a sub form inside the attach storage page.
+export class CreatePvcForm extends React.Component<CreatePvcFormProps, CreatePvcFormState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      storageClass: '',
+      pvcName: '',
+      accessMode: 'ReadWriteOnce',
+      requestSizeValue: '1',
+      requestSizeUnit: 'Gi',
+      disableForm: false,
+      showLabelSelectors: false,
+      nameValuePairs: [['', '']],
+      radios: [
+        {
+          value: 'ReadWriteOnce',
+          title: 'Single User (RWO)',
+        },
+        {
+          value: 'ReadWriteMany',
+          title: 'Shared Access (RWX)',
+        },
+        {
+          value: 'ReadOnlyMany',
+          title: 'Read Only (ROX)',
+        },
+      ],
+      dropdownUnits: {
+        Mi: 'MiB',
+        Gi: 'GiB',
+        Ti: 'TiB',
+      },
+    };
+  }
+
+  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    // this handles pvcName, accessMode, size
+    const { name, value } = event.currentTarget;
+    this.setState({ [name]: value } as any, () => {
+      this.onChange();
+    });
+  };
+
+  handleNameValuePairs = ({ nameValuePairs }) => {
+    this.setState({ nameValuePairs }, () => {
+      this.onChange();
+    });
+  };
+
+  handleStorageClass = storageClass => {
+    this.setState({ storageClass }, () => {
+      this.onChange();
+    });
+  };
+
+  handleClear = () => {
+    this.setState({ storageClass: '' }, () => {
+      this.onChange();
+    });
+  };
+
+  toggleShowLabelSelectors = () => {
+    this.setState({ showLabelSelectors: !this.state.showLabelSelectors });
+  };
+
+  onFormDisable = disableForm => {
+    this.setState({ disableForm });
+  };
+
+  nameValueArrayToObject = nvpArray => {
+    return _.reduce(nvpArray, (acc, [key, value]) => {
+      return key && value ? { ...acc, [key]: value } : acc;
+    }, {});
+  };
```

Comment:
Can you check if we already have a utility that does this? I'd expect we need it for the annotation modal.

Reply:
I have looked around but I have not found exactly what I need where the key is set as property name in that specific format to be passed.  Any more specific places to look?

Reply:
I'd look at

https://github.com/openshift/console/blob/3ed590323df8e59cdc0f09663696f7928abc090c/frontend/public/components/modals/tags.jsx

Reply:
I still don't see any method that is changing the array data into an object with key: value properties.  I think my function is concise and clean.  Can we leave this code as is?

Reply:
OK, but we should fix it to allow empty values. AFAIK, those shouldn't be skipped.

Reply:
Oh, I purposely removed the empty values.  It certainly makes no sense to try to save a label selector with no key.  I guess you could have a key with no value.  That would just require me to change: return key && value ? ... to just: return key  ? ....

Reply:
Right, key is required, but I think value can be empty

Reply:
Ok, I will change that after I run some test.